### DISCUSSION TITLE
Comparing Updated Form Data to Original

### DISCUSSION THREAD

**Discussion by jmfloyd4** [Budibase/budibase#5899] [General]
Hi there,
I'm writing an application where we would need to email through a summary of what was updated in a row. 

i.e. When updating a row, send an email saying: 
"Old First Name": Bob
"New First Name": Fred

I started doing this by making an Automation with an App Action. I was passing in each individual field's old value (from the data provider on the updatee) page to the automation. 

Then I was querying the updated row in the automation, and comparing the App Action field with the updated query. 

It's quite a big form, and there's plenty of them all different - so I'm hoping for a more efficient method than this? 
Is there a way to pass an entire Data Provider object with the old form data into an automation to compare? 

Appreciate any help. 
Thanks

**Comment by melohagan** [comment]
Hi @jmfloyd4 

You could try using a <a href="https://docs.budibase.com/docs/json">JSON Column</a> to store the rows that have been changed? 
After you send the email then you could wipe the data in that column.

**Reply by jmfloyd4** [reply]
Thanks so much! Perfect. I forgot about JSON Columns. This has enabled me to achieve what I'm after. Appreciate the help
